### PR TITLE
[Merge after #5]: Add Segmentation conversion

### LINKED ISSUE #2: [TODO]: Add Segmentation data
Shared with @weiglszonja who is working on this presently in conjunction with the other segmentation refactors on NeuroConv.

Two files: cells0.mat, cells1.mat.

From discussion:

> Inside each file cells0.mat are glial, cells1.mat are neurons. The variables are:
> freq: imaging frequency
> timeseries/baseline: detrended fluorescence and baseline
> x, y, z: coordinates of each cell (padded with zeros where needed)
> w: "spatial footprint" of each cell (sometimes useful for visualization)
> NB: the timeseries and baseline are stored in half-precision and may not load in python.

### How to map data to NWB

Each of the glial/neural cells will be written to different neurodata objects.

The combination of `x,y,z` values (each is a dataset of `number_of_cells x maximum_number_of_spanning_pixels` in Matlab). The zero-padding he refers to is with regards to that `maximum_number_of_spanning_pixels`, for example it might be a value of `164` but most cells might only take up `~15` pixels worth of space. Hence these values ought to be considered as the 'spatial weights' but will be boolean in that sense.

`w` has no real interpretation, it's just an abstract part of the analysis.

### Data specifics and questions

As they mentioned, the half-precision actually does create an issue for `h5py` for whatever reason - the datasets in question do not load their shape properly.

I'm saving alternate versions of these files to the same location but with adjusted data types - those are the ones we can use for the conversion of these fields.

This was done in MATLAB with

```matlab
baseline = cast(baseline, "single");
timeseries = cast(timeseries, "single");
save("{cell-name}_adjusted.mat", "baseline", "timeseries", "freq", "w", "x", "y", "z", "-v7.3")
```

and if we want to include it in the auto-convert workflow we could harness the Python MATLAB engine to do this for us as a pre-step.

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
WIP

Going to need some fixes to ROIExtractors and NeuroConv for volumetric image masks

### CODE REVIEW

File: src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        baseline_shape = self._file["baseline"].shape
+        assert baseline_shape != (1, 6,), (  # The MATLAB half-precision coding stores the dataset as a (1,6) object
+            "The flourescence series in this file are saved in half precision - file cannot be read in Python! "
+            "Please recast the MATLAB datatype and save a new file."
+        )
+
+        self._roi_response_raw = self._file["baseline"]
+        self._roi_response_dff = self._file["timeseries"]  # TODO: Notes describe this as 'detrended'
+        self._image_masks = 1
```
Comment (h-mayorquin) [inline src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py:33]: Is this necessary?

**Review by h-mayorquin — DISMISSED** [review]
Core structure looks good to me.

File: src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py:33]: It must have been but I don't recall why. Probably should instead be something fixed with ROIExtractors, will try to recall the issue

Comment (CodyCBakerPhD) [reply src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py:33]: Removed in #11

Comment (h-mayorquin) [reply src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py:33]: Great : )

**Pull request merged**